libraries/abstractions/transport/secure_sockets/transport_secure_sockets.c

```diff
@@ -0,0 +1,507 @@
+ * @param[in] recvTimeoutMs Timeout for transport recv.
+ *
+ * @return #SOCKETS_ERROR_NONE or error code from Secure Sockets.
+ */
+static int32_t transportTimeoutSetup( Socket_t tcpSocket,
+                                      uint32_t sendTimeoutMs,
+                                      uint32_t recvTimeoutMs );
+
+/**
+ * @brief Sets up a TLS session on top of a TCP connection using the Secure Sockets API.
+ *
+ * @param[out] pNetworkContext The output parameter to return the created network context.
+ * @param[in] pServerInfo Server connection info.
+ * @param[in] pSocketsConfig socket configs for the connection.
+ *
+ * @return #TRANSPORT_SOCKET_STATUS_SUCCESS on success;
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, #TRANSPORT_SOCKET_STATUS_INSUFFICIENT_MEMORY,
+ *         #TRANSPORT_SOCKET_STATUS_INVALID_CREDENTIALS, #TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR,
+ *         #TRANSPORT_SOCKET_STATUS_DNS_FAILURE, #TRANSPORT_SOCKET_STATUS_CONNECT_FAILURE on failure.
+ */
+static TransportSocketStatus_t establishConnect( NetworkContext_t * pNetworkContext,
+                                                 const ServerInfo_t * pServerInfo,
+                                                 const SocketsConfig_t * pSocketsConfig );
+
+/**
+ * @brief Set up TLS configurations for the socket.
+ *
+ * @param[in] pSocketsConfig The session configuration.
+ * @param[in] tcpSocket An initialized socket to secure.
+ * @param[in] pHostName Remote server name for SNI.
+ * @param[in] hostnameLength The length of `pHostName`.
+ *
+ * @return #SOCKETS_ERROR_NONE or #SOCKETS_SOCKET_ERROR.
+ */
+static int32_t tlsSetup( const SocketsConfig_t * pSocketsConfig,
+                         Socket_t tcpSocket,
+                         const uint8_t * pHostName,
+                         size_t hostnameLength );
+
+
+/*-----------------------------------------------------------*/
+
+int32_t SecureSocketsTransport_Send( NetworkContext_t * pNetworkContext,
+                                     const void * pMessage,
+                                     size_t bytesToSend )
+{
+    size_t bytesSent = 0;
+    int32_t TransportSocketStatus = SOCKETS_ERROR_NONE;
+    Socket_t xSocket = SOCKETS_INVALID_SOCKET;
+
+    if( ( pMessage == NULL ) ||
+        ( bytesToSend <= ( size_t ) 0 ) ||
+        ( pNetworkContext == NULL ) ||
+        ( pNetworkContext->pContext == NULL ) )
+    {
+        LogError( ( "TransportSendSecureSockets bad parameters pMessage %p, bytesToSend %d,  pvContext %p",
+                    pMessage, bytesToSend, pNetworkContext->pContext ) );
+        bytesSent = ( size_t ) SOCKETS_EINVAL;
+    }
+    else
+    {
+        xSocket = ( Socket_t ) pNetworkContext->pContext;
+        TransportSocketStatus = SOCKETS_Send( xSocket,
+                                              pMessage,
+                                              bytesToSend,
+                                              0 );
```

Comment:
Can the `xSocket` variable be avoided?
```suggestion
        TransportSocketStatus = SOCKETS_Send( ( Socket_t ) pNetworkContext->pContext;,
                                              pMessage,
                                              bytesToSend,
                                              0 );
```